src/tests/dotnet-counters/CounterMonitorPayloadTests.cs

```diff
@@ -0,0 +1,311 @@
+        public async Task TestCounterMonitorSystemRuntimeMetricsCSV(TestConfiguration configuration)
+        {
+            List<string> lines = await GetCounterTraceCSV(configuration, new List<string> { SystemRuntimeName });
+            IEnumerable<string[]> splitLines = lines.Select(l => l.Split(","));
+
+            ValidateSystemRuntimeMetrics(
+                splitLines.Select(line => line[1]).ToHashSet(),
+                splitLines.Select(line => line[2]).ToHashSet(),
+                splitLines.Select(line => line[3]).ToHashSet());
+        }
+
+        private void ValidateSystemRuntimeMetrics(ISet<string> actualProviders, ISet<string> actualCounterNames, IEnumerable<string> actualCounterTypes)
+        {
+            string[] ExpectedProviders = { "System.Runtime" };
+            Assert.Equal(ExpectedProviders, actualProviders);
+
+            // Could also just check the number of counter names
+            HashSet<string> expectedCounterNames = new()
+            {
+                "CPU Usage (%)",
+                "Working Set (MB)",
+                "GC Heap Size (MB)",
+                "Gen 0 GC Count (Count / 1 sec)",
+                "Gen 1 GC Count (Count / 1 sec)",
+                "Gen 2 GC Count (Count / 1 sec)",
+                "ThreadPool Thread Count",
+                "Monitor Lock Contention Count (Count / 1 sec)",
+                "ThreadPool Queue Length",
+                "ThreadPool Completed Work Item Count (Count / 1 sec)",
+                "Allocation Rate (B / 1 sec)",
+                "Number of Active Timers",
+                "GC Fragmentation (%)",
+                "GC Committed Bytes (MB)",
+                "Exception Count (Count / 1 sec)",
+                "% Time in GC since last GC (%)",
+                "Gen 0 Size (B)",
+                "Gen 1 Size (B)",
+                "Gen 2 Size (B)",
+                "LOH Size (B)",
+                "POH (Pinned Object Heap) Size (B)",
+                "Number of Assemblies Loaded",
+                "IL Bytes Jitted (B)",
+                "Number of Methods Jitted",
+                "Time spent in JIT (ms / 1 sec)"
+            };
+
+            Assert.Subset(actualCounterNames, expectedCounterNames);
+
+            string[] ExpectedCounterTypes = { Metric, Rate };
+            Assert.Equal(ExpectedCounterTypes, actualCounterTypes);
+        }
+
+        private async Task<JSONCounterTrace> GetCounterTraceJSON(TestConfiguration configuration, List<string> counterList)
+        {
+            string path = Path.ChangeExtension(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()), "json");
+
+            try
+            {
+                await GetCounterTrace(configuration, counterList, path, CountersExportFormat.json);
+                using FileStream metricsFile = File.OpenRead(path);
+                JSONCounterTrace trace = JsonSerializer.Deserialize<JSONCounterTrace>(metricsFile, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
```

Comment:
Let's make sure that casing is the same across all serializations (ours vs. original dotnet-counters) even if the deserializer is case insensitive